scripts/test-import-reborn-run-artifact.py

```
@@ -115,11 +172,193 @@ def artifact(messages: list[dict[str, object]]) -> dict[str, object]:
+            candidate["_review"]["required_actions"][0],
+        )
+        self.assertNotIn("failed", str(candidate))
+
+    def test_thread_artifact_rejects_other_unscoped_messages(self) -> None:
+        artifact = self.thread_artifact(
+            {"sequence": 1, "kind": "system", "status": "finalized", "content": "prompt"},
+            {"sequence": 2, "run_id": "run-1", "kind": "user", "content": "retry"},
+            {"sequence": 3, "run_id": "run-1", "kind": "assistant", "content": "done"},
+        )
+
+        with self.assertRaisesRegex(ValueError, "no replayable user message"):
```

Ruff (0.16.0)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)